### ISSUE TITLE
T2048.v is currently broken

### ISSUE THREAD

**Issue by artagnon** [ejgallego/coq-lsp#137]
The server crashes just before the entire file is checked.

**Comment by artagnon** [comment]
Logs aren't helpful here at all.

```
[Trace - 11:49:23] Received notification '$/coq/fileProgress'.
[Trace - 11:49:23] Received notification '$/coq/fileProgress'.
[Trace - 11:49:23] Received notification '$/coq/fileProgress'.
...
[Info  - 11:49:25] Connection to server got closed. Server will restart.
[Trace - 11:49:25] Sending request 'initialize - (0)'.
[Trace - 11:49:25] Received notification 'window/logMessage'.
[Info  - 11:49:25] Server started
[Trace - 11:49:25] Received response 'initialize - (0)' in 170ms.
[Trace - 11:49:25] Sending notification 'initialized'.
[Trace - 11:49:25] Sending notification 'textDocument/didOpen'.
[Trace - 11:49:25] Sending request 'textDocument/documentSymbol - (1)'.
[Trace - 11:49:25] Received response 'textDocument/documentSymbol - (1)' in 401ms. Request failed: Document is not ready (-32802).
[Trace - 11:49:25] [check]: resuming, from: 0 l: 1
```

**Comment by Alizter** [comment]
Is it freezing or exiting?

**Comment by artagnon** [comment]
I've narrowed down the crash to `Coq.Interp.interp ~st ~fb_queue` at the `Qed.` statement.

**Comment by ejgallego** [comment]
Don't we have a backtrace?

**Comment by ejgallego** [comment]
Would be also helpful to be able to bisect when this got broken.

**Comment by ejgallego** [comment]
How much memory and time does coqc take with the file?

**Comment by ejgallego** [comment]
Works fine here, but needs almost 2GiB of Ram and the Qed takes almost a minute to check; I think this is not specific to coq-lsp

**Comment by Alizter** [comment]
I also am able to run it but barely. I don't think there is an issue on our side.

**Comment by artagnon** [comment]
Not sure how to debug this further: we don't have backtraces. The crash seems platform-specific: I'm on macOS.

**Comment by Alizter** [comment]
Perhaps it would be possible to run coq-lsp in the CI? We should just listen to it's outputs, and perhaps set up our own dummy editor lsp requests. Doing something like this would let us easily see platform specific issues in the CI.

**Comment by ejgallego** [comment]
> Not sure how to debug this further: we don't have backtraces. The crash seems platform-specific: I'm on macOS.

Attach `gdb` to the running `coq-lsp` process. If you find a problem with that I'll be happy to help in person.

**Comment by artagnon** [comment]
Fixed by #182.

**Comment by artagnon** [comment]
No, memory wasn't the issue. I think the parser was crashing on macOS, and I think updating the submodule fixed it.
